code/modules/surgery/organs/lungs.dm

```diff
@@ -323,6 +323,36 @@
 			H.reagents.add_reagent(/datum/reagent/stimulum, max(0, 5 - existing))
 		breath.adjust_moles(GAS_STIMULUM, -gas_breathed)
 
+	// Carbon Monoxide
+		var/carbon_monoxide_pp = PP(breath,GAS_CO)
+		if (carbon_monoxide_pp > gas_stimulation_min)
+			H.reagents.add_reagent(/datum/reagent/carbon_monoxide, 1)
+			var/datum/reagent/carbon_monoxide/monoxide_reagent = H.reagents.has_reagent(/datum/reagent/carbon_monoxide)
+			if(!monoxide_reagent)
+				H.reagents.add_reagent(/datum/reagent/carbon_monoxide,2)
+			switch(carbon_monoxide_pp)
+				if (0 to 20)
+					monoxide_reagent.accumilation = min(monoxide_reagent.accumilation,50)
+				if (20 to 100)
+					monoxide_reagent.accumilation = min(monoxide_reagent.accumilation, 150)
+					H.reagents.add_reagent(/datum/reagent/carbon_monoxide,2)
+				if (100 to 200)
+					monoxide_reagent.accumilation = min(monoxide_reagent.accumilation, 250)
+					H.reagents.add_reagent(/datum/reagent/carbon_monoxide,4)
+				if (200 to 400)
+					monoxide_reagent.accumilation = min(monoxide_reagent.accumilation, 250)
+					H.reagents.add_reagent(/datum/reagent/carbon_monoxide,8)
+				if (400 to INFINITY)
+					monoxide_reagent.accumilation = max(monoxide_reagent.accumilation, 450)
+					H.reagents.add_reagent(/datum/reagent/carbon_monoxide,16)
```

Comment:
the way this scales could probably be done with math instead of a switch statment